src/components/media-views.js

```diff
@@ -733,7 +738,10 @@ AFRAME.registerComponent("media-image", {
           throw new Error(`Unknown image content type: ${contentType}`);
         }
 
-        textureCache.set(src, texture);
+        const cacheItem = textureCache.set(src, texture);
```

Comment:
`GIFTexture` extends `THREE.Texture` and writes:
```
    this.image.width = frames[0].width;
    this.image.height = frames[0].height;
```
Are GIFs being scaled correctly?

Reply:
It doesn't use TextureLoader to load gifs, so the texture cleanup after upload change doesn't apply to them (it couldn't anyway since the images are actually needed on the CPU side). The width/height here are width/height of the frame data provided by the gif worker.